packages/neuron-ui/src/components/PricePanel/index.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { localNumberFormatter, shannonToCKBFormatter } from 'utils'
+import RingProgressBar from 'widgets/RingProgressBar'
+import TextField from 'widgets/TextField'
+import { Transfer } from 'widgets/Icons/icon'
+import DropdownWithCustomRender from 'widgets/DropdownWithCustomRender'
+import { useTranslation } from 'react-i18next'
+import { appState, useState as useGlobalState } from 'states'
+
+import { FeeRateValueArrayItemType, useGetBatchGeneratedTx } from 'components/Send/hooks'
+import { DEFAULT_COUNT_DOWN } from 'utils/const'
+
+import styles from './pricePanel.module.scss'
+
+interface PricePanelProps {
+  field: string
+  price: string
+  onPriceChange: (value: string) => void
+}
+enum PriceTypeEnum {
+  Custom = 'custom',
+  Standard = 'standard',
+}
+const DEFAULT_PRICE_ARRAY = ['1000', '2000', '3000']
+
+const PricePanel: React.FunctionComponent<PricePanelProps> = ({ price, field, onPriceChange }: PricePanelProps) => {
+  const [t] = useTranslation()
+  const [type, setType] = useState<PriceTypeEnum>(PriceTypeEnum.Standard)
+  const [dropdownValue, setDropdowValue] = useState(DEFAULT_PRICE_ARRAY[1])
+  const [feeRateValueArray, setFeeRateValueArray] = useState<FeeRateValueArrayItemType[]>([])
+  const [priceArray, setPriceArray] = useState(DEFAULT_PRICE_ARRAY)
+  const [currentModeIdx, setCurrentModeIdx] = useState<number>(1)
+  const [inputError, setInputError] = useState<string | null>(null)
+  const {
+    app: {
+      send = appState.send,
+      feeRateStats: { suggestFeeRate = 0 },
+      countDown,
+    },
+    wallet: { id: walletID = '' },
+  } = useGlobalState()
+
+  const isStandard = type === PriceTypeEnum.Standard
+  const label = isStandard ? t('price-switch.price') : t('price-switch.customPrice')
+  const percent = (countDown / DEFAULT_COUNT_DOWN) * 100
+
+  const handleDropdownChange = useCallback(
+    (e: { value: string; index?: number }) => {
+      const { value: dropdownChangedValue, index = 1 } = e
+      setDropdowValue(dropdownChangedValue)
+      setCurrentModeIdx(index)
```

Comment:
Can we only save the mode idx, and create the dropdown value by idx and `priceArray`
Example:
When the user selects 'slow', if `priceArray` is [1000, 2000, 5000], the dropdown value will be `{key: 'slow', label: '1000'}`, and if `priceArray` is [2000, 4000, 10000], the dropdown value will be `{key: 'slow', label: '2000'}`.